Runs: Rewrite Runs section

ngrayluna opened the pull request:
## Description

Classic Noah, started off as a simple "update images" task...then grew to a complete Runs rewrite...here's a summary of what I did:

  New Documentation Pages (7 added):
  - customize-run-display.mdx - How to customize run display settings
  - delete-runs.mdx - Instructions for deleting runs
  - run-identifiers.mdx - Explanation of run IDs and identifiers
  - run-states.mdx - Documentation on run states
  - search-runs.mdx - How to search for runs
  - stop-runs.mdx - How to stop runs
  - view-logged-runs.mdx - Viewing logged run data

  Renamed File:
  - multiple-runs-per-process.mdx → initialize-run.mdx

  Major Edits:
  - models/runs.mdx - Significantly reduced (597 lines removed) as content was split out
  - models/runs/grouping.mdx - Reorganized and updated
  - models/runs/forking.mdx - Expanded with more content
  - models/runs/filter-runs.mdx - Updates and improvements
  - Updated navigation in docs.json

  Image Cleanup:
  - Removed 17 outdated images/GIFs
  - Added 6 new, updated images

<!-- Uncomment and add a description of the change -->


<!-- Optionally, uncomment the heading and add details about how you tested the change and how reviewers should test it. For example:
## Testing
- [ ] Local build succeeds without errors (`mint dev`)
- [ ] Local link check succeeds without errors (`mint broken-links`)
- [ ] PR tests succeed

Replace the `[ ]` with `[x]` to check off the item instead of leaving it unchecked.

Otherwise, delete this entire section from opening to closing comment.
-->

<!-- Optionally, uncomment the heading and add one or more lines like these. Otherwise, delete this entire section from opening to closing comment.
--> 

## Related issues

- Fixes https://wandb.atlassian.net/browse/DOCS-1716

ngrayluna committed:
Up to stop run section

ngrayluna committed:
refactoring runs overview page

ngrayluna committed:
Refactored run identifiers page

ngrayluna committed:
rearranged topics. First draft of forked runs

ngrayluna committed:
Fork a run draft

ngrayluna committed:
Updated fork a run description

ngrayluna committed:
WIP

github-actions[bot] committed:
Optimised images with calibre/image-actions

ngrayluna committed:
wip

ngrayluna committed:
Rename markdown file name

github-actions[bot] committed:
Optimised images with calibre/image-actions

ngrayluna committed:
Rename file, hashtags

ngrayluna committed:
small edits

ngrayluna committed:
fix links

ngrayluna committed:
removed old gifs.

ngrayluna committed:
Fixed grouping page

github-actions[bot] committed:
Optimised images with calibre/image-actions

ngrayluna committed:
updated images for runs, consolidate info

github-actions[bot] committed:
Optimised images with calibre/image-actions

ngrayluna committed:
change titles

github-actions[bot] committed:
Optimised images with calibre/image-actions

ngrayluna committed:
removed old gif

ngrayluna committed:
removed unused images

ngrayluna committed:
removed outdated images

ngrayluna committed:
Merge branch 'main' into update_run_overview_page

ngrayluna committed:
removed more old images from ko and ja

ngrayluna committed:
word smithing, removed old gifs

ngrayluna committed:
more updates on run identifiers

ngrayluna committed:
minor edits

ngrayluna committed:
fix broken link

mdlinville commented on models/automations/automation-events.mdx:
```diff
@@ -34,15 +34,15 @@ This section describes the events related to an artifact that can trigger an aut
 
 ### Run events
-An automation can be triggered by a change in a [run's status](/models/#run-states) or a change in a [metric value](/models/track/log/#what-data-is-logged-with-specific-wb-api-calls).
+An automation can be triggered by a change in a [run's status](/models/(/models/runs/run-states) or a change in a [metric value](/models/track/log/#what-data-is-logged-with-specific-wb-api-calls).
```
Munged URL? One more instance in this file.

mdlinville commented on models/runs/customize-run-display.mdx:
```diff
@@ -0,0 +1,96 @@
+---
+title: View all runs a project
```
```suggestion
title: View all runs a project
sidebarTitle: View all runs
```
I think it will improve the left nav to shorten some of these page nav titles.

mdlinville commented on models/runs/customize-run-display.mdx:
```diff
@@ -0,0 +1,96 @@
+---
+title: View all runs a project
+description: This section shows how to customize how runs are displayed in your project's runs table
```
```suggestion
description: Details about customizing how runs are displayed in your project's runs table
```

mdlinville commented on models/runs/customize-run-display.mdx:
```diff
@@ -0,0 +1,96 @@
+A workspace is limited to displaying a maximum of 1000 runs, regardless of its configuration.
+</Note> */}
+
+View all runs logged to your W&B project in the **Runs** tab of your project sidebar. The *Runs table* shows details about all of your runs in a project. Use the Runs table to compare runs, sort runs by specific columns, and organize runs into groups.
```
To connect the dots, is it worth mentioning that the Runs Table is in the Runs Tab, and that the Runs Table and Workspace runs list share a config?

mdlinville commented on models/runs/customize-run-display.mdx:
```diff
@@ -0,0 +1,96 @@
+    <img src="/images/runs/run_table_all.png" alt="Runs table"  />
+</Frame>
+
+## Add columns
```
These sections could be consolidated as H3s in a "Manage columns" section, all the way through Hide Columns. This would help the right-hand TOC.

mdlinville commented on models/runs/delete-runs.mdx:
```diff
@@ -0,0 +1,62 @@
+2. Select the **Runs** tab.
+3. Select the checkbox next to the runs you want to delete.
+4. Choose the **Delete** button (trash can icon) above the table.
+5. From the modal that appears, choose **Delete**.
```
Nit: I think we tend to use "drawer" instead of "modal".

mdlinville commented on models/runs/delete-runs.mdx:
```diff
@@ -0,0 +1,62 @@
+5. From the modal that appears, choose **Delete**.
+
+<Note>
+Once a run with a specific ID is deleted, its ID may not be used again. Trying to initiate a run with a previously deleted ID will show an error and prevent initiation.
```
```suggestion
A run ID cannot be reused, even after the run is deleted. Instead, the run will fail with an error.
```
I'd consider demoting both of these notes from notes, to avoid note stacking.

mdlinville commented on models/runs/delete-runs.mdx:
```diff
@@ -0,0 +1,62 @@
+For projects that contain a large number of runs, you can use either the search bar to filter runs you want to delete using Regex or the filter button to filter runs based on their status, tags, or other properties.
+</Note>
+
+### Run deletion workflow
```
What about a title like "Run deletion flowchart"? Otherwise the title reads a bit like "This is what you should do to delete runs".

mdlinville commented on models/runs/delete-runs.mdx:
```diff
@@ -0,0 +1,62 @@
+**Important**
+
+When you delete a run and choose to delete associated artifacts, the artifacts are permanently removed and can't be recovered, even if the run is restored later. This includes models linked to the Model Registry.
+</Note>
```
```suggestion
<Warning>
When you delete a run and choose to delete associated artifacts, the artifacts are permanently removed and can't be recovered, even if the run is restored later. This includes models linked to the Model Registry.
</Warning>
```

mdlinville commented on models/runs/filter-runs.mdx:
```diff
@@ -19,60 +16,37 @@ Filter runs based on their tags with the filter button.
 3. From left to right, select `"Tags"` from the dropdown menu, select a logic operator, and select a filter search value.
 
+<Frame>
+    <img src="/images/app_ui/filter_runs.gif" alt="Filter runs by tags"  />
```
Curious why the image is formatted this way. Could use a Markdown image and could set the alt text as a caption on the frame. https://www.mintlify.com/docs/components/frames#captions

mdlinville commented on models/runs/filter-runs.mdx:
```diff
@@ -19,60 +16,37 @@ Filter runs based on their tags with the filter button.
-## Export runs table to CSV
 
-Export the table of all your runs, hyperparameters, and summary metrics to a CSV with the download button.
+The expression editor shows a list of options for each term using autocomplete on column names and logical predicate structure. You can connect multiple logical predicates into one expression using "and" or "or" (and sometimes parentheses).
```
```suggestion
The expression editor shows a list of options for each term. Column names and logical predicate structures are completed automatically as you type. You can connect multiple logical predicates into one expression using "and" or "or" (and sometimes parentheses).
```

mdlinville commented on models/runs/forking.mdx:
```diff
@@ -1,68 +1,104 @@
 </Warning>
 
-Use `fork_from` when you initialize a run with [`wandb.init()`](/models/ref/python/functions/init) to "fork" from an existing W&B run. When you fork from a run, W&B creates a new run using the `run ID` and `step` of the source run.
+Explore different parameters or models from a specific point in an experiment without impacting the original run. To do this, "fork" from an existing W&B run. When you fork from a run, W&B creates a new run based on the the source run's `run ID` and `step`. After you initialize a forked run, you can continue logging to the new run independently from the source run.
```
Not convinced about "fork". It's what we are calling it, I think we can avoid the quotes.

mdlinville commented on models/runs/forking.mdx:
```diff
@@ -1,68 +1,104 @@
+{/* Use `fork_from` when you initialize a run with [`wandb.init()`](/models/ref/python/functions/init) to "fork" from an existing W&B run. When you fork from a run, W&B creates a new run using the `run ID` and `step` of the source run. */}
 
 <Note>
 * Forking a run requires [`wandb`](https://pypi.org/project/wandb/) SDK version >= 0.16.5
```
This is a very old SDK version at this point, can we lose this requirement?

mdlinville commented on models/runs/forking.mdx:
```diff
@@ -1,68 +1,104 @@
 
 <Note>
 * Forking a run requires [`wandb`](https://pypi.org/project/wandb/) SDK version >= 0.16.5
 * Forking a run requires monotonically increasing steps. You can not use non-monotonic steps defined with [`define_metric()`](/models/ref/python/experiments/run#define_metric) to set a fork point because it would disrupt the essential chronological order of run history and system metrics.
```
```suggestion
* Forking a run requires monotonically increasing steps. Attempting to set a fork point with non-monotonic steps defined with [`define_metric()`](/models/ref/python/experiments/run#define_metric) is not supported, because it would disrupt the essential chronological order of run history and system metrics.
```

mdlinville commented on models/runs/forking.mdx:
```diff
@@ -1,68 +1,104 @@
 * Forking a run requires monotonically increasing steps. You can not use non-monotonic steps defined with [`define_metric()`](/models/ref/python/experiments/run#define_metric) to set a fork point because it would disrupt the essential chronological order of run history and system metrics.
 </Note>
 
+Use the `fork_from` parameter in [`wandb.init()`](/models/ref/python/functions/init) to fork from an existing run. Specify the source run's unique `run ID` and the `step` you want to start the forked run from. 
```
Curious why it's showing here but commented out in line 12?

mdlinville commented on models/runs/forking.mdx:
```diff
@@ -1,68 +1,104 @@
+## Fork from a previously logged run
 
-To fork a run, use the `fork_from` argument in [`wandb.init()`](/models/ref/python/functions/init) and specify the source `run ID` and the `step` from the source run to fork from:
+The following code snippet demonstrates how to fork from a run that you previously logged to W&B. This might occur if you want to fork from a run in a different script or notebook. Or if someone else logged the source run that you want to fork from.
```
```suggestion
The following code snippet shows how to fork from a run that was previously logged to W&B. Use this approach to fork from a run in a different script or notebook, or from a run logged by a different user or service account.
```

mdlinville commented on models/runs/forking.mdx:
```diff
@@ -1,68 +1,104 @@
 
-## Continue from a forked run
-After initializing a forked run, you can continue logging to the new run. You can log the same metrics for continuity and introduce new metrics. 
+Note that you use the `original_run.id` property to obtain the unique run ID of the original run.
```
```suggestion
Use the `original_run.id` property to obtain the unique run ID of the original run.
```

mdlinville commented on models/runs/forking.mdx:
```diff
@@ -1,68 +1,104 @@
-After initializing a forked run, you can continue logging to the new run. You can log the same metrics for continuity and introduce new metrics. 
+Note that you use the `original_run.id` property to obtain the unique run ID of the original run.
+
+## Example script
```
```suggestion
### Example script
```

mdlinville commented on models/runs/grouping.mdx:
```diff
@@ -1,133 +1,146 @@
 ---
 
-Group individual jobs into experiments by passing a unique **group** name to **wandb.init()**.
+Organize your runs into *groups*. A group is a collection of runs that share a common purpose, such as training runs for a specific model or evaluation runs for a specific dataset.
```
What about something like "This page shows how to organize your runs using groups or using properties like [Job types](/models/runs/grouping#organize-runs-by-job-type) to indicate the function of a run, such as `preprocessing`, `training`, or `evaluation`."

mdlinville commented on models/runs/grouping.mdx:
```diff
@@ -1,133 +1,146 @@
-### 3. Toggle grouping by columns in the UI
+<Tabs>
+<Tab title="Python SDK">
+Programmatically add one or more runs to a group with the W&B Python SDK. Pass the name of your group as an argument to the `group` parameter when you initialize a run with `wandb.init(group="")`. The name you provide is used to organize and filter runs in the W&B App.
```
```suggestion
Programmatically add one or more runs to a group with the W&B Python SDK. Pass the name of your group as an argument to the `group` parameter when you initialize a run with `wandb.init(group="")`.  You can use group names to organize and filter runs in the W&B App.
```

mdlinville commented on models/runs/grouping.mdx:
````diff
@@ -1,133 +1,146 @@
+```
 
-### 4. Manage groups in the UI
+Within the project's workspace, you can view runs organized by group. The following image shows the runs table grouped by the group name. Three groups named `A`, `B`, and `C` appear in the runs table, each containing three runs.
````
```suggestion
In the project's workspace, you can view runs organized by group. The following image illustrates organizing the runs table by group name. Three groups named `A`, `B`, and `C` appear in the runs table, each containing three runs.
```

mdlinville commented on models/runs/grouping.mdx:
```diff
@@ -1,133 +1,146 @@
+3. Above the list of runs, click the **Group** button.
+4. Click the checkbox next to one or more runs you want to group.
+5. Select **Move to group**.
+6. In the modal, select an existing group or create a new group.
```
```suggestion
6. In the drawer, select an existing group or create a new group.
```

mdlinville commented on models/runs/grouping.mdx:
```diff
@@ -1,133 +1,146 @@
+
+1. In your project sidebar, select the **Runs** tab.
+2. Above the list of runs, click the **Group** button.
+3. From the dropdown, select either **Group**.
```
```suggestion
3. From the dropdown, select a **Group**.
```

mdlinville commented on models/runs/grouping.mdx:
```diff
@@ -1,133 +1,146 @@
+
+### Move runs between groups
+
+Move runs in one group to another group:
```
```suggestion
Move runs from one group to another:
```

mdlinville commented on models/runs/grouping.mdx:
```diff
@@ -1,133 +1,146 @@
+1. Within the modal, select the target group or create a new group.
+1. Click **Move**.
+
+### Remove a group
```
This title could use some disambiguation from the one on line 91. This is something like "Remove runs from a group" or "Stop grouping by a group"?

mdlinville commented on models/runs/grouping.mdx:
```diff
@@ -1,133 +1,146 @@
 
-To sort the list of runs by any visible column:
+<Note>
+W&B stores job types as a run [`wandb.Run.job_type`](/models/ref/python/experiments/run#property-run-job-type) property.
```
```suggestion
Specify the job type in the run's [`wandb.Run.job_type`](/models/ref/python/experiments/run#property-run-job-type) property.
```

mdlinville commented on models/runs/grouping.mdx:
```diff
@@ -1,133 +1,146 @@
+Organize runs by their *job type*. A job type indicates the function of a run, such as `preprocessing`, `training`, or `evaluation`. 
 
-To sort the list of runs by any visible column:
+<Note>
```
Not sure this needs to be a note or if the sentence is necessary at all. It seems to be covered in line 102.

mdlinville commented on models/runs/grouping.mdx:
````diff
@@ -1,133 +1,146 @@
+```
 
-Suppose you set grouping in `wandb.init()`, we will group runs by default in the UI. You can toggle this on and off by clicking the **Group** button at the top of the table. Here's an [example project](https://wandb.ai/carey/group-demo?workspace=user-carey) generated from [sample code](https://wandb.me/grouping) where we set grouping. You can click on each "Group" row in the sidebar to get to a d ...
+The following image shows runs organized by the job type:
````
```suggestion
The following image shows runs organized by job type:
```

mdlinville commented on models/runs/grouping.mdx:
```diff
@@ -1,133 +1,146 @@
+
+1. In your project sidebar, select the **Runs** tab.
+2. Above the list of runs, click the **Group** button.
+3. From the dropdown, select either **Job Type**.
```
```suggestion
3. From the dropdown, select **Job Type**.
```

mdlinville commented on models/runs/grouping.mdx:
Stopped review after this file, for now.

mdlinville commented on models/runs/manage-runs.mdx:
```diff
@@ -1,41 +1,51 @@
-
-If you group runs into experiments, refer to [Set a group in the UI](/models/runs/grouping/#set-a-group-in-the-ui).
-
 ## Move runs between your projects
```
How about section titles like "Move a run to a different project" and "Move a run to a different team"? I now you can move multiple runs at once, but I think you can't move multiple runs to different projects / teams at once, and the "between" is not the most important part because you probably moving only in one direction at a time. WDYT? Though as noted in my comment around line 22, I'm not sure this really needs to be presented as two different headings at all.

mdlinville commented on models/runs/manage-runs.mdx:
```diff
@@ -1,41 +1,51 @@
 2. Select the **Runs** tab from the project sidebar.
 3. Select the checkbox next to the runs you want to move.
-4. Choose the **Move** button above the table.
+4. Choose the **Move to project** button above the table.
```
```suggestion
4. Click the **Move to project** button above the table.
```
Not sure about choosing a button.

mdlinville commented on models/runs/manage-runs.mdx:
```diff
@@ -1,41 +1,51 @@
-</Frame>
-
-## Move runs to a team
+## Move runs between your teams
```
Unless I'm missing something, this is still moving a run to a new project, right? It just happens to be a project owned by a different team. This heading structure may give a reader the idea that you could move a run to a different team without moving it to a project owned by that team.

mdlinville commented on models/runs/rewind.mdx:
To avoid note stacking, consider reworking this as a Requirements or Prerequisites section?

mdlinville commented on models/runs/rewind.mdx:
I'd move this paragraph up to the very top, above the warning. Then the reader can see what the page is about before dealing with the private preview warning and the requirements.

mdlinville commented on models/runs/rewind.mdx:
```diff
@@ -78,7 +77,7 @@ run2.finish()
 
 
-After you rewind a run, you can explore archived run with the W&B App UI. Follow these steps to view archived runs:
+After you rewind a run, you can explore archived run with the W&B App. Follow these steps to view archived runs:
```
```suggestion
After you rewind a run, you can explore the original archived run in the W&B App. Follow these steps to view an archived run:
```

mdlinville commented on models/runs/run-identifiers.mdx:
```diff
@@ -0,0 +1,181 @@
+---
+title: Run IDs and names
```
How about "Find and customize a run's ID or name" or something more task oriented?

mdlinville commented on models/runs/run-identifiers.mdx:
```diff
@@ -0,0 +1,181 @@
+description: Learn how to find a run's unique identifier and run name, how to create a custom run ID, and how to customize a run's name.
+---
+
+When you initialize a W&B Run, W&B assigns that run a [unique identifier known as a *run ID*](/models/runs/run-identifiers#run-id). Each run also has a human-readable [non-unique *run name*](/models/runs/run-identifiers#run-name) that you can customize.
```
```suggestion
When you initialize a W&B Run, W&B gives it a [unique identifier known as a *run ID*](/models/runs/run-identifiers#run-id). Each run also has a human-readable [non-unique *run name*](/models/runs/run-identifiers#run-name) that you can customize.
```

mdlinville commented on models/runs/run-identifiers.mdx:
```diff
@@ -0,0 +1,181 @@
+
+## Run ID
+
+Run IDs uniquely identify runs. By default, W&B generates a [random and unique run ID](#autogenerated-run-ids) for you when you initialize a new run. You can also [specify your own unique run ID](#create-a-custom-run-id) when you [initialize a run](/models/runs/initialize-run). 
```
```suggestion
A run's ID uniquely identifies the run. By default, W&B generates a [random and unique run ID](#autogenerated-run-ids) automatically when you initialize a new run, unless you [specify your own unique run ID](#create-a-custom-run-id) when you [initialize the run](/models/runs/initialize-run). 
```

mdlinville commented on models/runs/run-identifiers.mdx:
```diff
@@ -0,0 +1,181 @@
+
+Run IDs uniquely identify runs. By default, W&B generates a [random and unique run ID](#autogenerated-run-ids) for you when you initialize a new run. You can also [specify your own unique run ID](#create-a-custom-run-id) when you [initialize a run](/models/runs/initialize-run). 
+
+### Find a run's unique ID
```
```suggestion
### Find a run's ID
```

mdlinville commented on models/runs/run-identifiers.mdx:
```diff
@@ -0,0 +1,181 @@
+
+### Find a run's unique ID
+
+W&B returns the unique run ID in the terminal when you initialize a run. For example, consider the following code snippet that initializes a W&B run:
```
```suggestion
When you initialize a run, W&B returns the unique run ID in the terminal. For example, consider the following code snippet that initializes a W&B run:
```

mdlinville commented on models/runs/run-identifiers.mdx:
```diff
@@ -0,0 +1,181 @@
+
+The last part of the run URL (`1jx1ud12`) is the unique run ID.
+
+You can also find a run's unique ID in the W&B App:
```
This makes me wonder if it is worth using a tabset for this, maybe "Python SDK" or "W&B App" or something?

mdlinville commented on models/runs/run-identifiers.mdx:
```diff
@@ -0,0 +1,181 @@
+</Frame>
+
+<Tip>
+Use a run's unique ID to directly navigate to that run's overview page in the W&B App. The following cell shows the URL path for a specific run:
```
Cell?

mdlinville commented on models/runs/run-identifiers.mdx:
````diff
@@ -0,0 +1,181 @@
+https://wandb.ai/<entity>/<project>/<run-id>
+```
+
+Where values enclosed in angle brackets (`< >`) are placeholders for the actual values of the entity, project, and run ID.
````
```suggestion
Replace values enclosed in angle brackets (`< >`) with the actual values of the entity, project, and run ID.
```

mdlinville commented on models/runs/run-identifiers.mdx:
```diff
@@ -0,0 +1,181 @@
+### Autogenerated run IDs
+
+W&B generates a random run ID for you if you do not specify a run ID when you initialize a run.
+
```
Is this stray? It's already mentioned above and there are no specific instructions in here.

mdlinville commented on models/runs/run-identifiers.mdx:
```diff
@@ -0,0 +1,181 @@
+
+Each run has a human-readable, non-unique run name. By default, W&B generates a random run name when you initialize a new run if you do not specify a run name for it. The name of a run appears within your project's workspace and at the top of the [run's **Overview** page](#overview-tab).
+
+Continuing from the previous example, the name of the run is `glwoing-shadows-8`.
```
```suggestion
Continuing from the previous example, the name of the run is `glowing-shadows-8`.
```

mdlinville commented on models/runs/run-identifiers.mdx:
```diff
@@ -0,0 +1,181 @@
+    <img src="/images/runs/unique-run-id.png" alt="Run ID location"  />
+</Frame>
+
+You can name your run when [you initialize it](/models/runs/run-identifiers#create-a-custom-run-name) or [rename](/models/runs/run-identifiers#rename-a-run) it after initializing it.
```
```suggestion
You can name your run when [you initialize it](/models/runs/run-identifiers#create-a-custom-run-name) or [rename](/models/runs/run-identifiers#rename-a-run) it afterward.
```

mdlinville commented on models/runs/run-identifiers.mdx:
```diff
@@ -0,0 +1,181 @@
+
+### Rename a run
+
+Rename a run after initializing it programmatically with Python SDK or interactively with the W&B App.
```
```suggestion
Rename a run after initializing it programmatically with the Python SDK or interactively in the W&B App.
```

mdlinville commented on models/runs/run-identifiers.mdx:
```diff
@@ -0,0 +1,181 @@
+<Tabs>
+<Tab title="Python SDK">
+
+Use [`wandb.Api.Run`](/models/ref/python/public-api/api#method-api-run) to access a run logged to W&B. This returns a [run object](/models/ref/python/public-api/run#property-run-name) that you can use to update the run name. Call `wandb.Api.Run.update()` method to persist changes.
```
```suggestion
Use [`wandb.Api.Run`](/models/ref/python/public-api/api#method-api-run) to access a run logged to W&B. This method returns a [run object](/models/ref/python/public-api/run#property-run-name) that you can use to update the run name. Call `wandb.Api.Run.update()` method to persist changes.
```

mdlinville commented on models/runs/run-identifiers.mdx:
```diff
@@ -0,0 +1,181 @@
+
+Use [`wandb.Api.Run`](/models/ref/python/public-api/api#method-api-run) to access a run logged to W&B. This returns a [run object](/models/ref/python/public-api/run#property-run-name) that you can use to update the run name. Call `wandb.Api.Run.update()` method to persist changes.
+
+Ensure to replace the values enclosed in angle brackets (`< >`) with your own values:
```
```suggestion
Replace the values enclosed in angle brackets (`< >`) with your own values.
```

mdlinville commented on models/runs/run-identifiers.mdx:
```diff
@@ -0,0 +1,181 @@
+Each run also has a *run display name* that you can customize for each workspace. The display name defaults to the same value as the run name. The display name appears in the run's workspace and runs table. 
+
+<Note>
+If you change a run's display name in one workspace, it does not change the run's display name in other workspaces or projects.
```
```suggestion
If you change a run's display name in one workspace, the display name changes only for that workspace, not in other workspaces or projects.
```

mdlinville commented on models/runs/run-identifiers.mdx:
```diff
@@ -0,0 +1,181 @@
+If you change a run's display name in one workspace, it does not change the run's display name in other workspaces or projects.
+</Note>
+
+Use the run display name to override the run name displayed in that workspace without renaming the run in the project.
```
I'd move this up to the end of the previous paragraph before the note. Then the note makes more sense.

mdlinville commented on models/runs/run-identifiers.mdx:
```diff
@@ -0,0 +1,181 @@
+
+Use the run display name to override the run name displayed in that workspace without renaming the run in the project.
+
+### Rename a display name
```
```suggestion
### Rename a run's display name
```

mdlinville commented on models/runs/run-states.mdx:
```diff
@@ -0,0 +1,70 @@
+---
+title: Run states
+description: Learn about the different states a W&B run can have and what they mean.
```
```suggestion
description: Learn about the different states a W&B run can have.
```

mdlinville commented on models/runs/run-states.mdx:
```diff
@@ -0,0 +1,70 @@
+description: Learn about the different states a W&B run can have and what they mean.
+---
+
+[Run states](/models/runs/run-states#run-states) indicate the current status of a W&B run. You can [view the state](/models/runs/run-states#view-the-state-of-a-run) of a run in the W&B App or programmatically using the W&B Python SDK.
```
```suggestion
The [Run state](/models/runs/run-states#run-states) indicate the current status of a W&B run. You can [view the state](/models/runs/run-states#view-the-state-of-a-run) of a run in the W&B App or programmatically using the W&B Python SDK.
```
Previous sections showed Python first, but it's flipped here. Intentional?

mdlinville commented on models/runs/search-runs.mdx:
```diff
@@ -0,0 +1,16 @@
+---
+title: "Search runs"
+description: Learn how to search runs using regular expressions (RegEx) and export the runs table to CSV in the W&B App.
```
```suggestion
description: Learn how to search runs using regular expressions and export the runs table to CSV in the W&B App.
```
Suggest removing it from the description because it's also in the first paragraph, and it's also not a broadly agreed-upon abbreviation (there is also RegExp, regex, reg-ex, etc etc).

mdlinville commented on models/runs/search-runs.mdx:
```diff
@@ -0,0 +1,16 @@
+
+## Search End Time for runs
+
+We provide a column named `End Time` that logs that last heartbeat from the client process. The field is hidden by default.
```
```suggestion
## Search for runs by end time

The **End Time** column logs logs the last heartbeat from the client process. The column is hidden by default.
```
Link to docs for showing a hidden column?

mdlinville commented on models/runs/stop-runs.mdx:
```diff
@@ -0,0 +1,46 @@
+description: Learn how to stop runs programmatically using the W&B Python SDK or manually from the W&B App.
+---
+
+Stop a run programmatically with the W&B Python SDK or manually from the W&B App.
```
```suggestion
Stop a run programmatically with the W&B Python SDK or interactively in the W&B App.
```

mdlinville commented on models/runs/stop-runs.mdx:
```diff
@@ -0,0 +1,46 @@
+
+<Tabs>
+<Tab title="Python SDK">
+1. Navigate to the terminal or code editor where you initialized the run.
+2. Press `Ctrl+D` to stop the run.
```
```suggestion
In the terminal or code editor where you initialized the run, press `Ctrl+D` to stop the run.
```

mdlinville commented on models/runs/stop-runs.mdx:
````diff
@@ -0,0 +1,46 @@
+wandb: Find logs at: ./wandb/run-20241106_095857-o8sdbztv/logs
+```
+
+Navigate to the W&B App to confirm the run is no longer active:
````
This is weird. So you actually need to use the App whether you are doing it programmatically or interactively? Seems a little silly. If the run is no longer active, wouldn't the terminal have returned already?

ngrayluna replied:
Not necessarily. Sweeps (for e.g.) can seem finished, but they're not. There are other times, too when a run "hangs", e.g. if you don't use a context manager and don't explicitly call run.finish() in a notebook.

mdlinville commented on models/runs/stop-runs.mdx:
```diff
@@ -0,0 +1,46 @@
+
+1. Navigate to the project that your run was logging to.
+2. Select the name of the run. 
+  <Note>
```
Again, this is weird. If you are in the app we are directing you to the terminal, and above in the terminal instructions we tell you to check the app first.

mdlinville commented on models/runs/tags.mdx:
```diff
@@ -72,7 +72,7 @@ This method is best suited to applying a tag or tags to a single run manually.
 ## Remove tags from one or more runs
 
-Tags can also be removed from runs with the W&B App UI.
+Tags can also be removed from runs with the W&B App.
```
```suggestion
Follow these steps to remove tags from a run in the W&B App.
```

mdlinville commented on models/runs/view-logged-runs.mdx:
```diff
@@ -0,0 +1,137 @@
+description: Learn how to view a specific logged run and its properties using the W&B App or the LEET terminal UI.
+---
+
+View information about a specific run such as the state of the run, artifacts logged to the run, metrics logged to a run, system metrics, and more.
```
```suggestion
View information about a specific run, such as its current state, artifacts, metrics, and more.
```

mdlinville commented on models/runs/view-logged-runs.mdx:
````diff
@@ -0,0 +1,137 @@
+https://wandb.ai/<team-name>/<project-name>/runs/<run-id>
+```
+
+Where values enclosed in angle brackets (`< >`) are placeholders for the actual values of the team name, project name, and [run ID](/models/runs/run-identifiers#run-id).
````
```suggestion
Replace values enclosed in angle brackets (`< >`) with the actual values of the team name, project name, and [run ID](/models/runs/run-identifiers#run-id).
```

mdlinville commented on models/runs/view-logged-runs.mdx:
```diff
@@ -0,0 +1,137 @@
+
+Where values enclosed in angle brackets (`< >`) are placeholders for the actual values of the team name, project name, and [run ID](/models/runs/run-identifiers#run-id).
+
+After you select a run, explore its properties by navigating through the tabs: [Overview](/models/runs/view-logged-runs#overview), [Logs](/models/runs/view-logged-runs#logs), [Files](/models/runs/view-logged-runs#files), [Code](/models/runs/view-logged-runs#code), and [Artifacts](/models/runs/view-logged-runs#artifacts).
```
```suggestion
Explore the run's properties by navigating through the tabs: [Overview](/models/runs/view-logged-runs#overview), [Logs](/models/runs/view-logged-runs#logs), [Files](/models/runs/view-logged-runs#files), [Code](/models/runs/view-logged-runs#code), and [Artifacts](/models/runs/view-logged-runs#artifacts).
```

mdlinville commented on models/runs/view-logged-runs.mdx:
```diff
@@ -0,0 +1,137 @@
+</Tabs>
+
+
+### Overview
```
I'm not sure but I think that all of these H3s relate only to the W&B App tab above, not the LEET tab. Can you confirm?

ngrayluna replied:
Confirming. It's odd. Maybe I should split the tabs and create new sections (W&B App and LEET, respectively). Then nest the current H3s into that? 🤔

mdlinville commented on models/runs.mdx:
Now reviewed up to here.

mdlinville commented on models/runs.mdx:
```diff
@@ -6,18 +6,31 @@ title: Overview
 A *run* is a single unit of computation logged by W&B. You can think of a W&B Run as an atomic element of your whole project. In other words, each run is a record of a specific computation, such as training a model and logging the results, hyperparameter sweeps, and so forth.
 
-Common patterns for initiating a run include, but are not limited to: 
+Common reasons for initiating a run include, but are not limited to: 
```
```suggestion
Some common reasons for initiating a run include: 
```
But this seems like kind of a weird way to talk about this.

mdlinville commented on models/runs/initialize-run.mdx:
I think you need a redirect for this

mdlinville reviewed (commented):
Here's the rest of the first pass review. :)

ngrayluna committed:
WIP fixing suggestions

ngrayluna committed:
more review edits

ngrayluna committed:
more feedback

ngrayluna committed:
customize run display

ngrayluna committed:
condense move runs

ngrayluna committed:
added redirect

ngrayluna committed:
Merge branch 'main' into update_run_overview_page

ngrayluna commented:
> Here's the rest of the first pass review. :)

Thanks so much for the review. Addressed comments. Left a couple of comments and/or questions on a couple of them. :)

mdlinville commented on models/automations/automation-events.mdx:
```diff
@@ -34,15 +35,15 @@ This section describes the events related to an artifact that can trigger an aut
 
 ### Run events
-An automation can be triggered by a change in a [run's status](/models/(/models/runs/run-states) or a change in a [metric value](/models/track/log/#what-data-is-logged-with-specific-wb-api-calls).
+An automation can be triggered by a change in a [run's status](/models/runs/run-states) or a change in a [metric value](/models/track/log/#what-data-is-logged-with-specific-wb-api-calls).
```
Nit: Omit the trailing slash after `log` to avoid an unnecessary redirect in the MIntlify SaaS environment.

mdlinville commented on models/runs/customize-run-display.mdx:
```diff
@@ -1,21 +1,25 @@
 </Note> */}
 
-View all runs logged to your W&B project in the **Runs** tab of your project sidebar. The *Runs table* shows details about all of your runs in a project. Use the Runs table to compare runs, sort runs by specific columns, and organize runs into groups.
+View all runs logged to your W&B project in the **Runs** tab of your project sidebar. Within the Runs tab is the *Runs table*. The Runs table shows details about all of your runs in a project. Use the Runs table to compare runs, sort runs by specific columns, and organize runs into groups.
```
What about "Within the Runs tab, the Runs table shows...."

mdlinville commented on models/runs/customize-run-display.mdx:
```diff
@@ -1,21 +1,25 @@
-## Add columns
+## Manage columns
+
+The following sections describe how to add, remove, move, pin, hide, and sort columns in the Runs table.
```
What about "how to customize the Runs table"?

mdlinville commented on models/runs/delete-runs.mdx:
```diff
@@ -55,8 +55,6 @@ graph TB
-When you delete a run and choose to delete associated artifacts, the artifacts are permanently removed and can't be recovered, even if the run is restored later. This includes models linked to the Model Registry.
-</Note>
\ No newline at end of file
+<Warning >
```
Nit: Extra space

mdlinville commented on models/runs/forking.mdx:
```diff
@@ -7,13 +7,11 @@ title: Fork a run
-Explore different parameters or models from a specific point in an experiment without impacting the original run. To do this, "fork" from an existing W&B run. When you fork from a run, W&B creates a new run based on the the source run's `run ID` and `step`. After you initialize a forked run, you can continue logging to the new run independently from the source run.
-
-{/* Use `fork_from` when you initialize a run with [`wandb.init()`](/models/ref/python/functions/init) to "fork" from an existing W&B run. When you fork from a run, W&B creates a new run using the `run ID` and `step` of the source run. */}
+Explore different parameters or models from a specific point in an experiment without impacting the original run. To do this, *fork* from an existing W&B run. When you fork from a run, W&B creates a new run based on the the source run's `run ID` and `step`. After you initialize a forked run, you can continue logging to the new run independently from the source run.
```
What about "To explore....., *fork* from an existing...."

mdlinville commented on models/runs/forking.mdx:
```diff
@@ -7,13 +7,11 @@ title: Fork a run
 <Note>
 * Forking a run requires [`wandb`](https://pypi.org/project/wandb/) SDK version >= 0.16.5
-* Forking a run requires monotonically increasing steps. You can not use non-monotonic steps defined with [`define_metric()`](/models/ref/python/experiments/run#define_metric) to set a fork point because it would disrupt the essential chronological order of run history and system metrics.
+* Forking a run requires monotonically increasing steps. Attempting to set a fork point with non-monotonic steps defined with [`define_metric()`](/models/ref/python/experiments/run#define_metric) is not supported, because it would disrupt the essential chronological order of run history and system metrics.
```
Nit: Consider replacing some of these gerunds.

mdlinville commented on models/runs/run-identifiers.mdx:
```diff
@@ -134,15 +138,17 @@ run.update()
 Use the run display name to override the run name displayed in that workspace without renaming the run in the project.
 
-### Rename a display name
+### Rename a run's display name
```
"Rename a .... name" seems funny. How about "Change a run's display name"?

mdlinville commented on models/runs/run-states.mdx:
````diff
@@ -53,6 +53,9 @@ for run in runs:
     print()
 ```
+
+You can 
````
Stray?

mdlinville commented on models/runs/search-runs.mdx:
```diff
@@ -1,15 +1,14 @@
 ---
 title: "Search runs"
-description: Learn how to search runs using regular expressions (RegEx) and export the runs table to CSV in the W&B App.
+description: Learn how to search runs using regular expressions and export the runs table to CSV in the W&B App.
```
Am I missing something? I don't see anything aboiut export in this file.

mdlinville commented on models/runs/stop-runs.mdx:
```diff
@@ -1,42 +1,26 @@
-Stop a run programmatically with the W&B Python SDK or manually from the W&B App.
+Stop a run programmatically with the W&B Python SDK or interactively in the W&B App.
+
+<Note tip>
```
?

mdlinville commented on models/runs.mdx:
```diff
@@ -6,7 +6,7 @@ title: Overview
 A *run* is a single unit of computation logged by W&B. You can think of a W&B Run as an atomic element of your whole project. In other words, each run is a record of a specific computation, such as training a model and logging the results, hyperparameter sweeps, and so forth.
 
-Common reasons for initiating a run include, but are not limited to: 
+Some common reasons for initiating a run include: 
```
I keep feeling like this is a strange phrasing. Maybe "Start a run to: - Train a model...."

mdlinville reviewed (approved):
A few nits and tiny questions, approved so as not to block.

ngrayluna committed:
resolve merge conflicts

ngrayluna committed:
feedback

ngrayluna committed:
Updated filter and search run pages

mdlinville committed:
Trigger link checker re-run with corrected workflow

This change triggers a new deployment which will use the corrected link
checker workflow. The previous run was using an older version from main
that incorrectly checked against docs.wandb.ai instead of the preview URL.

mdlinville committed:
Add PR commenting to link checker workflow

This adds functionality to automatically comment on PRs with link checker
results when the Mintlify deployment completes. The comment will be updated
on subsequent runs and clearly indicates whether all links are valid or if
issues were found.

mdlinville committed:
Fix syntax errors in link checker PR comment step

- Fix JavaScript syntax error when exit_code is empty
- Handle both pull_request and deployment_status events properly
- Use safe string interpolation for exit code
- Fallback to docs.wandb.ai when no deployment URL available

mdlinville committed:
Merge remote-tracking branch 'origin/main' into update_run_overview_page

ngrayluna committed:
Merge branch 'main' into update_run_overview_page

ngrayluna commented:
Merging branch despite messy error message. As noted in this [thread](https://weightsandbiases.slack.com/archives/C081R4TDAH1/p1768592739745899?thread_ts=1768506183.884029&cid=C081R4TDAH1) the reports added to this PR are incorrect.
